p2phttp.go

```diff
@@ -57,14 +57,51 @@ import (
 	protocol "github.com/libp2p/go-libp2p-protocol"
 )
 
-// P2PProtocol is used to tag and identify streams
+// DefaultP2PProtocol is used to tag and identify streams
 // handled by go-libp2p-http
-var P2PProtocol protocol.ID = "/libp2p-http"
+var DefaultP2PProtocol protocol.ID = "/libp2p-http"
+
+// Options holds configuration options for the transport.
+type Options struct {
```

Comment:
The [functional options pattern](https://github.com/grpc/grpc-go/blob/4dedfdc82cb6616457887f5ab38e2b0c0bb20875/server.go#L112) usually makes the backing struct unexported so that the only way for it to be updated is by the provided option functions, i.e. `ProtocolOption()`.

Reply:
Ok, I had doubts about how Godoc shows exporting types involving unexported types. But if that's the default..